includes/managers/controls.php

```diff
@@ -436,7 +436,17 @@ public static function get_controls_names() {
 	private function register_controls() {
 		$this->controls = [];
 
-		foreach ( self::get_controls_names() as $control_id ) {
+		$controls_names = self::get_controls_names();
+
+		if ( ! Plugin::instance()->experiments->is_feature_active( 'container_grid' ) ) {
```

Comment:
maybe extract the logic to an external function 
should be something like :
if (! Plugin::instance()->experiments->is_feature_active( 'container_grid' )){
 unsetGapsIfNotUsed()
}